exchanges/okx/okx_wrapper.go

```diff
@@ -216,27 +214,32 @@ func (ok *Okx) Setup(exch *config.Exchange) error {
 		OrderbookBufferConfig: buffer.Config{
 			Checksum: ok.CalculateUpdateOrderbookChecksum,
 		},
-	})
-
-	if err != nil {
+	}); err != nil {
 		return err
 	}
-	err = ok.Websocket.SetupNewConnection(stream.ConnectionSetup{
+
+	go ok.WsReadData()
+	go ok.WsResponseMultiplexer.Run()
+
+	if err := ok.Websocket.SetupNewConnection(stream.ConnectionSetup{
 		URL:                  okxAPIWebsocketPublicURL,
 		ResponseCheckTimeout: exch.WebsocketResponseCheckTimeout,
 		ResponseMaxLimit:     okxWebsocketResponseMaxLimit,
 		RateLimit:            500,
-	})
-	if err != nil {
+	}); err != nil {
 		return err
 	}
-	return ok.Websocket.SetupNewConnection(stream.ConnectionSetup{
+	if err := ok.Websocket.SetupNewConnection(stream.ConnectionSetup{
```

Comment:
Sometimes I avoid tailing a err return, because it's not the intrinsic return value of a func.
It's no more meaningful than the previous SetupNewConnection.
This way it's clear for future maintainers that if they want to add something that happens "at the end" or "close to the end" they don't put it above this SetupNewConn, or feel scared to turn `return funcCall` back into an if statement.

I'm not committed to it until death, but it's deliberate and reasoned. 😄 

Have I swayed you?

Reply:
Okay, I can change this then. Will do so now.